lib/java/test/org/apache/thrift/TestStruct.java

```diff
@@ -309,11 +309,33 @@ public void testToString() throws Exception {
   }
 
   public void testBytesBufferFeatures() throws Exception {
-    JavaTestHelper o = new JavaTestHelper();
+    
+    final String testString = "testBytesBufferFeatures";
+    final JavaTestHelper o = new JavaTestHelper();
+
     o.setReq_bin((ByteBuffer)null);
     assertNull(o.getReq_bin());
+
+    o.setReq_bin(ByteBuffer.wrap(testString.getBytes()));
+    assertNotNull(o.getReq_bin());
```

Comment:
Yes, couldn't hurt.  Really I was just trying to test to make sure it doesn't trigger a ClassCastException; but your suggestion seems logical.